### ISSUE TITLE
Keyboard can't reconnect after extension gets enabled

### ISSUE THREAD

**Issue by berkakkaya** [maniacx/Bluetooth-Battery-Meter#36]
I was testing this extension in Ubuntu 24.04.1 LTS with Gnome 46 Wayland session. Extension shows battery percentage of my keyboard (Logitech Pebble Keys 2 K380s) perfectly but when I close and reopen my keyboard from its switch, it begins to have issues connecting to my computer. The only way I was able to fix the issue is to restart the system entirely. This issue disappears when I disable the extension.

**Comment by maniacx** [comment]
How are you trying to connect your device? 
1. Use Gnome-shell quick settings panel?
2. Using Gnome-control-center (settings)?
3. Or using bluetoothctl?

Kindly do the following.

 1.Enable the extension?

 2. Turn your keyboard off and on again 

 3. Try connecting using Gnome-shell  settings. 

![image](https://github.com/user-attachments/assets/b08486f6-26f5-4aa0-8783-46a87997d0f1)

4. If it does not connect try connection using Gnome control center (settings). (Also copy the mac address shown here)
![image](https://github.com/user-attachments/assets/b32f266e-5543-4f37-958d-0bb6ab358a95)

5. If it still does not connect try connecting using bluetoothctl

Let me know which menthod works and which doesn't.

Also try the aboive steps with extension disabled.

This will give me an idea of what is going on.
Sometime i have issue when connecting or disconnecting my mouse using bluetoothctl with/without extension enabled, so if you have the same issue with bluetoothctl with extension disabled, skip the test .

Thank you.

**Comment by berkakkaya** [comment]
> How are you trying to connect your device?
>
> Use Gnome-shell quick settings panel?
> Using Gnome-control-center (settings)?
> Or using bluetoothctl?

I was trying to reconnect my device through quick settings panel. After that had failed, I tried to connect it through the settings app too.

I did some more testing, also tried steps that you have mentioned. Seems like the issue isn't related to this extension, because I've managed to reproduce the same issue without enabling this extension. As far as I observed, the issue appears when there's a audio playback to another Bluetooth headset while doing the reconnection. Also, I saw that device discovery in settings app gets broken too after the issue appears.

My guess is that the Bluetooth components in OS doesn't handle the situation well, or the OS doesn't properly communicate with hardware in this scenario. I'll look forward for more suitable place to report this bug. Thank you for your support and sorry for wasting your time a little. Have a wonderful day! :)
